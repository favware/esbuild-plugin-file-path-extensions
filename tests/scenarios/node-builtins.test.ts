import esbuild, { type BuildOptions } from 'esbuild';
import { assertFileContent, buildAbsolutePath, createEsbuildConfig } from '../common/util';

function createConfig(platform: 'node' | 'browser'): BuildOptions {
  return createEsbuildConfig({
    platform,
    format: 'esm',
    outExtension: {
      '.js': '.mjs'
    },
    entryPoints: [buildAbsolutePath('../fixtures/build-in/node.mts')]
  });
}

describe('Node builtins', () => {
  test('GIVEN platform = "browser" THEN adds extensions to builtins', async () => {
    const config = createConfig('browser');

    await esbuild.build(config);

    await assertFileContent('../fixtures/build-out/node.mjs');
  });

  test('GIVEN platform = "node" THEN does not add extensions to builtins', async () => {
    const config = createConfig('node');

    await esbuild.build(config);

    await assertFileContent('../fixtures/build-out/node.mjs');
  });
});
