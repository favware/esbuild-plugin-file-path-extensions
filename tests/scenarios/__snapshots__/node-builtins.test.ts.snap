// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Node builtins > GIVEN platform = "browser" THEN adds extensions to builtins 1`] = `
"// tests/fixtures/build-in/node.mts
import assert from "assert.mjs";
import fs from "node:fs.mjs";
import fsp from "node:fs/promises.mjs";
assert(true);
fs.readdirSync(".");
void fsp.readdir(".");
"
`;

exports[`Node builtins > GIVEN platform = "node" THEN does not add extensions to builtins 1`] = `
"// tests/fixtures/build-in/node.mts
import assert from "assert";
import fs from "node:fs";
import fsp from "node:fs/promises";
assert(true);
fs.readdirSync(".");
void fsp.readdir(".");
"
`;
